Makefile

```diff
@@ -1,6 +1,14 @@
 # Make hacks
 .INTERMEDIATE:
 
+# Library versioning, following libtool's 'version-info' scheme (see:
+# info '(libtool) Libtool versioning' or
+# https://www.gnu.org/software/libtool/manual/html_node/Libtool-versioning.html).
+LT_CURRENT := 0
+LT_REVISION := 0
+LT_AGE := 0
+LT_VERSION_INFO := $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
```

Comment:
Another idea is to use Semanitic Versioning and just follow the current release version. In my opinion it is more simple, but @LIJI32 may have his own preference.

https://semver.org/

Reply:
When using libtool's `-release`, it does that, but `info "(libtool) Link mode"` says:
> '-release RELEASE'
     Specify that the library was generated by release RELEASE of your
     package, so that users can easily tell what versions are newer than
     others.  Be warned that no two releases of your package will be
     binary compatible if you use this flag.  If you want binary
     compatibility, use the '-version-info' flag instead (*note
     Versioning::).

So `-version-info` with the recommended scheme seems a better approach, at least when using libtool.